core/base/src/main/java/alluxio/collections/ConcurrentIdentityHashMap.java

```diff
@@ -0,0 +1,347 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.collections;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public class ConcurrentIdentityHashMap<K, V> implements ConcurrentMap<K, V> {
+
+  private final ConcurrentHashMap<IdentityObject<K>, V> mInternalMap;
+
+  /**
+   * Creates a new, empty map with the default initial table size (16).
+   */
+  public ConcurrentIdentityHashMap() {
+    mInternalMap = new ConcurrentHashMap<>();
+  }
+
+  /**
+   * Creates a new, empty map with an initial table size
```

Comment:
nit: are you wrapping the linewidth too early? it does not look like 100 char for me

Reply:
I borrowed some of these descriptions from ConcurrentMap constructor. I will change them